backends/tc/backend.cpp

```diff
@@ -354,6 +354,32 @@ void ConvertToBackendIR::updateDefaultMissAction(const IR::P4Table *t, IR::TCTab
                 if (defaultActionProperty->isConstant) {
                     tabledef->setDefaultMissConst(true);
                 }
+                bool directionParamPresent = false;
+                auto paramList = actionCall->action->getParameters();
+                for (auto param : paramList->parameters) {
+                    if (param->direction != IR::Direction::None) directionParamPresent = true;
+                }
+                if (!directionParamPresent) {
+                    auto i = 0;
+                    for (auto param : paramList->parameters) {
+                        auto defaultParam = new IR::TCDefaultActionParam();
+                        defaultParam->setParamName(param->name.originalName);
+                        auto defaultArg = methodexp->arguments->at(i++);
+                        if (defaultArg->expression->is<IR::Constant>()) {
+                            auto constVal = defaultArg->expression->to<IR::Constant>();
+                            bool sign;
+                            if (const IR::Type_Bits *tb = constVal->type->to<IR::Type_Bits>()) {
+                                sign = tb->isSigned;
+                            } else {
+                                sign = false;
+                            }
+                            defaultParam->setDefaultValue(
+                                Util::toString(constVal->value, 0, sign, constVal->base));
+                        } else
+                            defaultParam->setDefaultValue(defaultArg->expression->toString());
```

Comment:
What are the possible expressions? Can we avoid using toString()?

Reply:
Updated the code